lib/SILOptimizer/Utils/InstOptUtils.cpp

```diff
@@ -1458,12 +1458,9 @@ bool swift::simplifyUsers(SingleValueInstruction *inst) {
     if (!svi)
       continue;
 
-    SILValue S = simplifyInstruction(svi);
-    if (!S)
-      continue;
-
-    replaceAllSimplifiedUsesAndErase(svi, S, callbacks);
-    changed = true;
+    callbacks.resetHadCallbackInvocation();
+    simplifyAndReplaceAllSimplifiedUsesAndErase(svi, callbacks);
```

Comment:
I am going to take a quick look around our usage of deBlocks here. I think the right behavior is to not optimize if we aren't passed a real deBlock and we need one.